test/src/validators/email_validator_test.dart

```diff
@@ -26,9 +26,9 @@ void main() {
     });
 
     test('FormControl invalid with not String email', () {
-      final control = FormControl(validators: [Validators.email]);
+      final control = FormControl<String>(validators: [Validators.email]);
```

Comment:
This will not allow creating a control with dynamic type as:
```dart
final control = FormControl(
   value: "I'm a String control",
   validators: [Validators.email], // exception is thrown
);
```

Or even create a control with a different custom type like:
```dart
final control = FormControl<CustomEmail>(
   value: CustomEmail("johndoe@gmail.com"),
   validators: [Validators.email], // exception is thrown
);
```
So I suggest leaving the Email validator as **dynamic**

Reply:
Are you sure that the second example is valid?
The `CustomEmail` should have correct `toString` override to make email validator work
Also second example demands correct value accessor to make TextInput work

Reply:
Yes, as you say the class should override toString and the widget should have a Custom Control Value Accessor

Reply:
I rewrite the comments  left